### ISSUE TITLE
Leaflet map module

### ISSUE THREAD

**Issue by psolymos** [PredictiveEcology/wbi_shiny#3]
This module implements a Leaflet map to show 1 element (given period and treatment).

Leaflet object with satellite background & linking to a specific tile set.

Include full-screen mode option.

**Comment by bradlindblad** [comment]
@psolymos are the tiles that we are loading from the server already processed with the spectral colors, and we're just matching the legend to those?

**Comment by psolymos** [comment]
@bradlindblad yes, the tiles are processed with the spectral color, and the range of color values (101 hex values in `pal`) map to the range of the values between 0 and the max of the 4 maps (2 scenarios x 2 time periods for a given element). So for each individual map, we need to check the max and adjust the palette. Here is what I am thinking (`app/R/fct_map.R`, note: `X` is data from `data/element-stats.csv`):

```R
  # If `add_legend = TRUE`, show the legend
  if (add_legend) {
    
    title <- "Abundance"   # title for the legend
    v <- X[X$element_name == element & X$scenario == scenario & X$year == year & X$resolution == "1000m",]
    
    # add legend to the leaflet object
    m <- m |>
      leaflet::addLegend(
        position = "bottomleft", 
        pal = leaflet::colorNumeric(
          palette = grDevices::hcl.colors(101, "spectral", rev = TRUE)[seq_len(v$pal_max)],
          domain = c(0, v$max)
        ), 
        values = c(0, v$max),
        title = title,
        opacity = opacity
      )
    
  }
```

**Comment by bradlindblad** [comment]
Roger, that's where I was headed, thanks @psolymos

**Comment by psolymos** [comment]
Closing this. Both maps look good now.

**Comment by mthomas-ketchbrook** [comment]
Hi @psolymos, just as an FYI, @bradlindblad and I had been working on the legends in a new branch called "legend" over the weekend.  It looks like you already tackled this on the main branch  I figured I would leave the "legend" branch out there for you to take a look at in case you find anything helpful.  Feel free to delete the branch when you are done with it.

**Comment by psolymos** [comment]
@mthomas-ketchbrook and @bradlindblad : thanks a lot! Very similar to what I had done except I used function args instead of referring to the data object.

I also made quite a few changes to allow deployment (e.g. region plots) which uncovered some issues/limitations with {golem} itself and its dependency {dockerfiler}. The live version as it stands now is deployed here: https://wbi-nwt.analythium.app/
